### PR TITLE
Prefer existing user name when updating user cache from commit

### PR DESCRIPTION
**Pull request by yufeih**
This makes github user cache stable before an record is expired. Fixes PR unexpected change caused by display name.

image

### CODE REVIEW

**Comment by OsmondJiang** [comment]
#3987

File: src/docfx/lib/github/GitHubUserCache.cs [hunk 1]
```diff
@@ -248,7 +248,7 @@ async Task<(Error error, bool collide)> SaveRemote(string content)
         /// 2. User missing with the specified login: { "login": "..." }
         /// 3. User missing with the specified email: { "emails": [ "..." ] }
         /// </summary>
-        private void UpdateUser(GitHubUser user)
+        private void UpdateUser(GitHubUser user, bool preferExistingName = false)
```
Comment (OsmondJiang) [inline src/docfx/lib/github/GitHubUserCache.cs]: when will it be false?

Comment (yufeih) [reply src/docfx/lib/github/GitHubUserCache.cs]: All other places where users are updated:
- Users returned from _get by login` API
- Users read from cache file itself

File: src/docfx/lib/github/GitHubUserCache.cs [hunk 2]
```diff
@@ -291,25 +291,25 @@ private void UpdateUser(GitHubUser user)
...
+        private static void MergeUser(GitHubUser user, GitHubUser existingUser, bool preferExistingName)
         {
             if (existingUser.IsValid())
             {
                 if (user.Id is null)
                     user.Id = existingUser.Id;
                 if (user.Login is null)
                     user.Login = existingUser.Login;
-                if (user.Name is null)
+                if (user.Name is null || preferExistingName)
```
Comment (superyyrrzz) [inline src/docfx/lib/github/GitHubUserCache.cs]: `if (user.Name is null || preferExistingName && existingUser.Name != null)` ?

**Pull request merged**